### PR TITLE
Feat/860 material UI incorporation

### PR DESCRIPTION
**Pull request by lasmil**
- closes https://github.com/nearform/mercurius-federation-info-graphiql-plugin/issues/86
- introduced material-ui
- because we started to use scss files, I continued on the same path

### CODE REVIEW

File: src/App/App.jsx [hunk 1]
```diff
@@ -0,0 +1,40 @@
...
+  return (
+    <div
+      className={
+        hideSecondPanel ? 'full-height hide-second-panel' : 'full-height'
+      }
+    >
+      <GraphiQL
+        fetcher={fetcher}
+        plugins={[federationInfoPlugin({ federationSchemaUrl })]}
+        onTogglePluginVisibility={plugin => {
```
Comment (matijagaspar) [inline src/App/App.jsx:28]: I don't think this approach is ok, unless we change how mercurious implements graphiql plugins.

See here: <https://github.com/mercurius-js/mercurius/blob/a230f81872418603874cdb7f4ac110be0231f353/static/main.js#L121>

The problem is that mercurious only adds plugins to `plugins=` attribute. So ideally would be if we can trigger that panel hiding from the plugin itself (through context maybe?)

Comment (lasmil) [reply src/App/App.jsx:28]: Hmm yes you are right. As discussed we will use this PR to incorporate MUI and a future PR for the full screen mode.

**Comment by andreaforni** [comment]
I'm getting this error when I run `npm run start` (it works on `master`): 

Screenshot 2022-12-02 at 16 19 30

File: package.json [hunk 2]
```diff
@@ -37,10 +37,14 @@
     "@babel/preset-react": "^7.18.6",
     "@commitlint/cli": "^17.2.0",
     "@commitlint/config-conventional": "^17.2.0",
+    "@emotion/react": "^11.10.5",
+    "@emotion/styled": "^11.10.5",
```
Comment (andreaforni) [inline package.json:41]: Is emotion used somewhere? I did a quick search, and looks like these two packages are not used.

Comment (lasmil) [reply package.json:41]: They need to come with material-ui based on the documentation. I guess some components are using them.

Comment (simoneb) [reply package.json:41]: they're part of the installation instructions of MUI

**Comment by lasmil** [comment]
> 

Weird, now I also managed to replicate this. I am checking it

File: package.json (hunk 2 above)
Comment (andreaforni) [reply package.json:41]: Ah, I see. Thanks for the clarification.

**Comment by andreaforni** [comment]
> I'm getting this error when I run `npm run start` (it works on `master`):
> 
> Screenshot_2022-12-02_at_16_19_30

Thanks to a call with @lasmil and @matijagaspar, we figured out how to fix that: we had to enable the rollup to replace plugin only at build time, in this way, it does not interfere with dev mode.

**Review by matijagaspar — APPROVED** [review]

**Pull request merged**